org.eclipse.xtext/src/org/eclipse/xtext/generator/trace/AbstractTraceRegionToString.java

```diff
@@ -0,0 +1,434 @@
+				render(child, idW, offsetW, lengthW, (indent + 2), result);
+			}
+			result.add((id + ": " + Strings.repeat(" ", loc.length())) + "}");
+		}
+	}
+
+	protected String render() {
+		File localFile = new File(null);
+		LinkedHashMap<SourceRelativeURI, File> remoteFiles = CollectionLiterals.newLinkedHashMap();
+		List<RegionHandle> roothandles = CollectionLiterals.newArrayList();
+		int maxid = collect(getTrace(), 1, localFile, remoteFiles, roothandles);
+		int idwidth = Integer.toString(maxid, radix).length();
+		addAll(localFile.lines, render(localFile, idwidth));
+		for (File file : remoteFiles.values()) {
+			addAll(file.lines, render(file, idwidth));
+		}
+		int localWidth = Math.max(max(map(localFile.lines, (String it) -> Integer.valueOf(it.length()))),
+				getLocalTitle().length() + 2);
+		int remoteWidth = max(map(remoteFiles.values(),
+				(File it) -> Math.max(max(map(it.lines, (String it_1) -> Integer.valueOf(it_1.length()))),
+						getRemoteTitle(it.uri).length() + 2)));
+		localFile.lines.add(0, title(null, localWidth));
+		for (File file_1 : remoteFiles.values()) {
+			file_1.lines.add(0, title(file_1.uri, remoteWidth));
+		}
+		List<String> left = localFile.lines;
+		List<String> right = toList(concat(map(remoteFiles.values(), it -> it.lines)));
+		ArrayList<String> result = CollectionLiterals.newArrayList();
+		if (showLegend) {
+			result.add(
+					"Regions are surrounded by [N[ ... ]N]. Regions on the left and right with the same N are associated.");
+		}
+		for (int i = 0; ((i < ((Object[]) Conversions.unwrapArray(left, Object.class)).length)
+				|| (i < ((Object[]) Conversions.unwrapArray(right, Object.class)).length)); i++) {
+			String l = Strings.padEnd(i < left.size() ? left.get(i) : "", localWidth, ' ');
+			String r = i < right.size() ? right.get(i) : "";
+			result.add(l + " | " + r);
+		}
+		if (showTree) {
+			result.add(Strings.repeat("-", localWidth + remoteWidth + 3));
+			if (showLegend) {
+				result.add(
+						"<N>: <isDebug> <offset>-<length> <RegionJavaClass> -> <LocationJavaClass>[<offset>,<length>,<uri>]");
+			}
+			Iterable<RegionHandle> allhandles = concat(map(roothandles, it -> collect(it, it_1 -> it_1.children)));
+			int offsetWidth = String
+					.valueOf(max(map(allhandles, (RegionHandle it) -> Integer.valueOf(it.region.getMyOffset()))))
+					.length();
+			int lengthWidth = String
+					.valueOf(max(map(allhandles, (RegionHandle it) -> Integer.valueOf(it.region.getMyLength()))))
+					.length();
+			for (RegionHandle handle : roothandles) {
+				render(handle, idwidth, offsetWidth, lengthWidth, 1, result);
+			}
+		}
+		return join(result, org.eclipse.xtext.util.Strings.newLine());
+	}
+
+	@Override
+	public String toString() {
+		String _xtrycatchfinallyexpression = null;
+		try {
+			return render();
+		} catch (Throwable t) {
+			if (t instanceof Exception) {
+				_xtrycatchfinallyexpression = Throwables.getStackTraceAsString(t);
+			} else {
+				throw Exceptions.sneakyThrow(t);
```

Comment:
More sneaky

Reply:
Not sure what to do with this ... any suggestion?